libcst/codemod/visitors/_imports.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from dataclasses import dataclass, replace
+from typing import Optional
+
+from libcst.helpers import get_absolute_module
+
+
+@dataclass(frozen=True)
+class ImportItem:
+    """Representation of individual import items for codemods."""
+
+    module_name: str
+    obj: Optional[str] = None
+    alias: Optional[str] = None
+    relative: int = 0
+
+    def __post_init__(self) -> None:
+        if self.module_name is None:
+            object.__setattr__(self, "module_name", "")
+        elif self.module_name.startswith("."):
+            mod = self.module_name.lstrip(".")
+            rel = self.relative + len(self.module_name) - len(mod)
+            object.__setattr__(self, "module_name", mod)
+            object.__setattr__(self, "relative", rel)
+
+    @property
+    def module(self) -> str:
+        return "." * self.relative + self.module_name
+
+    def resolve_relative(self, base_module: Optional[str]) -> "ImportItem":
+        """Return an ImportItem with an absolute module name if possible."""
+        mod = self
+        # `import ..a` -> `from .. import a`
+        if mod.relative and mod.obj is None:
+            mod = replace(mod, module_name="", obj=mod.module_name)
```

Comment:
Unfortunately `replace`'s first argument is also called `obj`, so this won't work (as shown by the unit tests)

Reply:
huh. any idea why the local tests passed? (perhaps because i only ran them on 3.9?)

Reply:
Yeah this only seems to be a problem with 3.6 because we use the dataclasses package from pypi, where `obj` is not a positional-only argument.

Reply:
See also https://bugs.python.org/issue37163